docs/reference/mapping/fields/synthetic-source.asciidoc

```diff
@@ -0,0 +1,170 @@
+[[synthetic-source]]
+==== Synthetic `_source`
+
+Though very handy to have around, the source field takes up a significant amount
+of space on disk. Instead of storing source documents on disk exactly as you
+send them, Elasticsearch can reconstruct source content on the fly. Enable this
+by setting `synthetic: true` in `_source`:
+
+[source,console,id=enable-synthetic-source-example]
+----
+PUT idx
+{
+  "mappings": {
+    "_source": {
+      "synthetic": true
+    }
+  }
+}
+----
+// TESTSETUP
+
+While this on the fly reconstruction is *generally* slower than saving the source
+documents verbatim and loading them at query time, it saves a lot of storage
+space. There are a couple of restrictions to be aware of:
+
+* When you retrieve sythetic `_source` content it undergoes minor
+<<synthetic-source-modifications,modifications>> compared to the original JSON.
+* Synthetic `_source` can be used with indices that contain only these field
+types:
+
+** <<boolean-synthetic-source,`boolean`>>
+** <<numeric-synthetic-source,`byte`>>
+** <<numeric-synthetic-source,`double`>>
+** <<numeric-synthetic-source,`float`>>
+** <<geo-point-synthetic-source,`geo_point`>>
+** <<numeric-synthetic-source,`half_float`>>
+** <<numeric-synthetic-source,`integer`>>
+** <<ip-synthetic-source,`ip`>>
+** <<keyword-synthetic-source,`keyword`>>
+** <<numeric-synthetic-source,`long`>>
+** <<numeric-synthetic-source,`scaled_float`>>
+** <<numeric-synthetic-source,`short`>>
+** <<text-synthetic-source,`text`>>
+
+[[synthetic-source-modifications]]
+===== Synthetic source modifications
+
+When synthetic `_source` is enabled, retrieved documents undergo some
+modifications compared to the original JSON.
+
+[[synthetic-source-modifications-alphabetical]]
+====== Alphabetical sorting
+Synthetic `_source` fields are sorted alphabetically. For example:
+
+[source,console,id=synthetic-source-sorted-example]
+----
+PUT idx/_doc/1
+{
+  "foo": 1,
+  "bar": 2,
+  "baz": 3
+}
+----
+// TEST[s/$/\nGET idx\/_doc\/1?filter_path=_source\n/]
+
+Will become:
+
+[source,console-result]
+----
+{
+  "bar": 2,
+  "baz": 3,
+  "foo": 1
+}
```

Comment:
reading this, as much as I understand why you mention it, keys ordering is never a guarantee in JSON land. I wonder if it's then needed to provide an example. Could we shorten it and say "You should never rely on keys ordering, but if you do beware you'll lose that with synthetic source as what you'll get back is not 100% what you sent"?

Reply:
I get that. I'm providing an example for everything so I like the.... balance? I don't know the right word. I could link to the spec and mention that ordering isn't supported by the spec.

Reply:
I don't have a strong opinion but while the examples with arrays ought to be mentioned, taking this one out would make the docs page a little more compact without losing much?

Reply:
We need a tie breaker. @romseygeek, break a tie!

Reply:
No pressure :)

Reply:
I'm with @javanna, we don't guarantee key ordering anywhere so I don't think an example is necessary here.

Reply:
Democracy wins. I'll remove the example. I'll reduce this to a note.